x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/query/LongScriptFieldDistanceFeatureQuery.java

```diff
@@ -0,0 +1,211 @@
+    public Weight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float boost) throws IOException {
+        return new Weight(this) {
+            @Override
+            public boolean isCacheable(LeafReaderContext ctx) {
+                return false;
+            }
+
+            @Override
+            public void extractTerms(Set<Term> terms) {}
+
+            @Override
+            public Scorer scorer(LeafReaderContext context) throws IOException {
+                return new DistanceScorer(this, leafFactory.apply(context), context.reader().maxDoc(), boost);
+            }
+
+            @Override
+            public Explanation explain(LeafReaderContext context, int doc) throws IOException {
+                AbstractLongScriptFieldScript script = leafFactory.apply(context);
+                script.runForDoc(doc);
+                long value = valueWithMinAbsoluteDistance(script);
+                float weight = LongScriptFieldDistanceFeatureQuery.this.boost * boost;
+                float score = score(weight, distanceFor(value));
+                return Explanation.match(
+                    score,
+                    "Distance score, computed as weight * pivot / (pivot + abs(value - origin)) from:",
+                    Explanation.match(weight, "weight"),
+                    Explanation.match(pivot, "pivot"),
+                    Explanation.match(origin, "origin"),
+                    Explanation.match(value, "current value")
+                );
+            }
+        };
+    }
+
+    private class DistanceScorer extends Scorer {
+        private final AbstractLongScriptFieldScript script;
+        private final TwoPhaseIterator twoPhase;
+        private final DocIdSetIterator disi;
+        private final float weight;
+
+        protected DistanceScorer(Weight weight, AbstractLongScriptFieldScript script, int maxDoc, float boost) {
+            super(weight);
+            this.script = script;
+            twoPhase = new TwoPhaseIterator(DocIdSetIterator.all(maxDoc)) {
+                @Override
+                public boolean matches() throws IOException {
+                    script.runForDoc(approximation().docID());
+                    return script.count() > 0;
+                }
+
+                @Override
+                public float matchCost() {
+                    return MATCH_COST;
+                }
+            };
+            disi = TwoPhaseIterator.asDocIdSetIterator(twoPhase);
+            this.weight = LongScriptFieldDistanceFeatureQuery.this.boost * boost;
+        }
+
+        @Override
+        public int docID() {
+            return disi.docID();
+        }
+
+        @Override
+        public DocIdSetIterator iterator() {
+            return disi;
+        }
+
+        @Override
+        public TwoPhaseIterator twoPhaseIterator() {
+            return twoPhase;
+        }
+
+        @Override
+        public float getMaxScore(int upTo) throws IOException {
+            return boost;
```

Comment:
Should a potential max score be:  `weight` instead of `boost`?  
This method and several other methods  don't seem to throw `IOException`

Reply:
Yes! good catch.